common/workunit/workunit.cpp

```diff
@@ -11779,13 +11779,11 @@ extern WORKUNIT_API void submitWorkUnit(const char *wuid, const char *username,
     MemoryBuffer buffer;
     Owned<INamedQueueConnection> conn = createNamedQueueConnection(0); // MORE - security token?
     Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
-    Owned<IWorkUnit> workunit = factory->updateWorkUnit(wuid);
-    assertex(workunit);
-    StringAttr clusterName(workunit->queryClusterName());
+    Owned<IConstWorkUnit> cw = factory->openWorkUnit(wuid);
```

Comment:
@jakesmith please review.  This is the actual fix.  All the other changes are a rename to make it clearer which functions are calling getWorkUnitFactory(), and which are directly accessing the cache.  We may want to revisit some of the others.

Reply:
Looks fine, I have no idea why this was calling updateWorkUnit before.